internal/endpoint/pop3/pop3.go

```diff
@@ -0,0 +1,466 @@
+	mailbox, err := endp.getMailbox(user)
+	if err != nil {
+		return nil, err
+	}
+
+	c := make(chan *imap.Message)
+	err = mailbox.ListMessages(false, nil, nil, c)
+	if err != nil && err != mess.ErrNoMessages {
+		return nil, err
+	}
+
+	items := make([]string, 0)
+	for msg := range(c) {
+		items = append(items, strconv.FormatUint(uint64(msg.Uid), 10))
+	}
+
+	return items, nil
+}
+
+// Similar to ListMessage, but returns unique ID by message ID instead of size.
+func (endp *Endpoint) UidlMessage(user pop3backend.User, msgId int) (exists bool, uid string, err error) {
+	mailbox, err := endp.getMailbox(user)
+	if err != nil {
+		return false, "", err
+	}
+
+	seqset := imap.SeqSet{}
+	seqset.AddNum(uint32(msgId))
+	c := make(chan *imap.Message)
+	err = mailbox.ListMessages(true, &seqset, nil, c)
+	if err != nil && err != mess.ErrNoMessages {
+		return false, "", err
+	}
+
+	msg, ok := <- c
+	if !ok {
+		return false, "", nil
+	}
+
+	return true, strconv.FormatUint(uint64(msg.Uid), 10), nil
+}
+
+// Write all changes to persistent storage, i.e. delete all messages marked as deleted.
+func (endp *Endpoint) Update(user pop3backend.User) error {
+	mailbox, err := endp.getMailbox(user)
+	if err != nil {
+		return err
+	}
+
+	return mailbox.Expunge()
+}
+
+// If the POP3 server issues a positive response, then the
+// response given is multi-line.  After the initial +OK, the
+// POP3 server sends the headers of the message, the blank
+// line separating the headers from the body, and then the
+// number of lines of the indicated message's body, being
+// careful to byte-stuff the termination character (as with
+// all multi-line responses).
+// Note that if the number of lines requested by the POP3
+// client is greater than than the number of lines in the
+// body, then the POP3 server sends the entire message.
+func (endp *Endpoint) Top(user pop3backend.User, msgId int, n int) (lines []string, err error) {
+	return nil, fmt.Errorf("pop3: unimplemented")
+}
+
+// Lock is called immediately after client is connected. The best way what to use Lock() for
+// is to read all the messages into cache after client is connected. If another user
+// tries to lock the storage, you should return an error to avoid data race.
+func (endp *Endpoint) Lock(user pop3backend.User) error {
+	return nil // FIXME: NOT IMPLEMENTED
+}
+
+// Release lock on storage, Unlock() is called after client is disconnected.
+func (endp *Endpoint) Unlock(user pop3backend.User) error {
+	backendUser, ok := user.(imapbackend.User)
+	if !ok {
+		return fmt.Errorf("internal server error")
+	}
+
+	return backendUser.Logout()
+}
+
+func (endp *Endpoint) I18NLevel() int {
+	be, ok := endp.Store.(i18nlevel.Backend)
+	if !ok {
+		return 0
+	}
+	return be.I18NLevel()
+}
+
+func (endp *Endpoint) SupportedThreadAlgorithms() []sortthread.ThreadAlgorithm {
+	be, ok := endp.Store.(sortthread.ThreadBackend)
+	if !ok {
+		return nil
+	}
+
+	return be.SupportedThreadAlgorithms()
+}
```

Comment:
Not sure if these methods are necessary.

Reply:
These are relevant for IMAP implementation only and can be safely removed.